src/darwin/Framework/CHIP/MTRCSRInfo.h

```diff
@@ -54,11 +57,41 @@ API_AVAILABLE(ios(16.4), macos(13.3), watchos(9.4), tvos(16.4))
  */
 @property (nonatomic, copy, readonly) NSData * attestationSignature;
 
+/**
+ * Initialize an MTROperationalCSRInfo by providing all the fields.  It's the
+ * caller's responsibility to ensure that the provided csr matches
+ * csrElementsTLV.
+ */
 - (instancetype)initWithCSR:(MTRCSRDERBytes)csr
                    csrNonce:(NSData *)csrNonce
              csrElementsTLV:(MTRTLVBytes)csrElementsTLV
        attestationSignature:(NSData *)attestationSignature;
 
+/**
+ * Initialize an MTROperationalCSRInfo by providing the csrNonce (for example,
+ * the nonce the client initially supplied), and the csrElementsTLV and
+ * attestationSignature that the server returned.
+ */
+- (instancetype)initWithCSRNonce:(NSData *)csrNonce
```

Comment:
Should the `init` variations that take both a nonce / CSR and a TLV ensure that the TLV matches the other parameters? I'm not sure what all the code paths where this class is used are, but the concern would be that we're making it easy to accidentally plug together the NOC issuing process in an unsafe way (that e.g. doesn't validate that "our" nonce was used)?

Reply:
@ksperling-apple Please take a look?

Reply:
Looks good, just need to make sure those init methods are declared as nullable... I guess technically this might be a breaking change for Swift clients :-(